clang/lib/StaticAnalyzer/Core/RangeConstraintManager.cpp

```diff
@@ -2836,6 +2836,10 @@ bool RangeConstraintManager::canReasonAbout(SVal X) const {
     return false;
   }
 
+  // Non-integer types are not supported.
+  if (X.getAs<nonloc::LazyCompoundVal>())
+    return false;
+
```

Comment:
My problem with this is that I think LCVs shouldn't even appear here.
And even if they could, why don't we also handle `nonloc::CompoundVals` too? They are basically the same thing, except for the lazyness.

So, this patch masks some fundamental problem somewhere else - probably within the iterator checker or container modeling, actually passing an LCV.
But I get it, its better to not crash, sure.

I'd be happy to merge this and creating a new ticket for investigating what's going on.